File: src/main/java/my/study/repository/design/pattern/state/java/Context.java
package my.study.repository.design.pattern.state.java;

/**
 * @author djh on  2018/11/12 22:06
 * @E-Mail 1544579459@qq.com
 */
public class Context {
    private State mState;

    public void setState(State state) {
        mState = state;
    }

    void controlToNextChannel() {
        mState.nextChannel();
    }

    void controlTurnUpVolume() {
        mState.turnUpVolume();
    }
}

File: src/main/java/my/study/repository/design/pattern/state/kotlin/Context.kt
package my.study.repository.design.pattern.state.kotlin

/**
 * @author djh on  2018/11/12 22:23
 * @E-Mail 1544579459@qq.com
 */
class Context {
    
    lateinit var state: State
    
    fun controlToNextChannel() {
        state.nextChannel()
    }
    
    fun contorlTurnUpVolume() {
        state.turnUpVolume()
    }
}